File: oneRev.js
const diet = {
	"breakfast": "eggs",
	"lunch": "eggs",
	"dinner": "bacon"
}


const rev = (foods) => {
	let spun = {};
	for (food in foods) {
		let oldVal = foods[food]
		if (oldVal in spun == false) {
			spun[oldVal] = [food];
		} else {
			spun[oldVal].push(food);
		}
	}
	return spun
	}
console.log(rev(diet))

File: one_rev.py
diet = {
        "breakfast": "eggs",
        "lunch": "eggs",
        "dinner": "pizza"
        }

def spin(stuff):
    spun = {}
    for item in stuff:
        old_value = stuff[item]
        if old_value not in spun:
            spun[old_value] = [item]
        else:
            spun[old_value].append(item)
    return spun
            
print(spin(diet))
